test/higher_order_ops/test_with_effects.py

```diff
@@ -532,6 +532,110 @@ def forward(self, tangents_1, tangents_2):
     return (clone, clone_1, tangents_1, tangents_2)""",
             )
 
+    @skipIfNoDynamoSupport
+    def test_effectful_op_in_backward(self):
+        with torch.library._scoped_library("_mylib", "FRAGMENT") as lib:
+            lib.define("foo(Tensor x) -> Tensor")
+
+            def foo_impl(a):
+                return a.clone()
+
+            def foo_bwd(ctx, grad):
+                return torch.ops._mylib.foo(grad)
```

Comment:
can you also have this op increment a counter? That way we can test that the counter was actually incremented once in the forward and again in the backward